### PR TITLE
chore(observability): update check-events script

### PR DESCRIPTION
**Pull request by jdrouet**
Replaces #11571 and #11572

This PR fixes the trace messages, ensure `error_type` and `stage` are using constants and that `errors` and `counter` tags are consistent.

### CODE REVIEW

**Comment by jdrouet** [comment]
I'm cherry picking the commits from #11572

File: scripts/check-events [hunk 1]
```diff
@@ -6,12 +6,32 @@ require 'find'
 # These members/tags are common to multiple events
 BYTE_SIZE_COUNT = ['byte_size', 'count']
 
+# SUFFIX => [MESSAGE, COUNTERS, ADDITIONAL_TAGS]
```
Comment (bruceg) [inline scripts/check-events:9]: I think there may have been some miscommunication here. The original version of this had an additional "required" item in this data. In the [component spec](https://github.com/vectordotdev/vector/blob/master/docs/specs/component.md#bytesreceived) these event structs have certain required fields that this was verifying. I think that behavior needs to be restored to ensure compliance with the spec.

File: src/sources/aws_sqs/source.rs [hunk 2]
```diff
@@ -83,7 +83,7 @@ impl SqsSource {
         let receive_message_output = match result {
             Ok(output) => output,
             Err(err) => {
-                error!("SQS receive message error: {:?}", err);
+                error!("SQS receive message error: {:?}.", err);
```
Comment (bruceg) [inline src/sources/aws_sqs/source.rs:86]: We should ideally not be using format strings in structured output, though `check-events` isn't scanning for this yet. We can tackle that in a separate issue.

File: scripts/check-events [hunk 3]
```diff
@@ -6,12 +6,32 @@ require 'find'
...
+  ],
+  'EventsReceived' => [
+    'Events received.', ['received_events', 'received_event_bytes'], ['count', 'byte_size']
+  ],
+  'EventsSent' => [
+    'Events sent.', ['sent_events', 'sent_event_bytes'], ['count', 'byte_size']
+  ],
+  'BytesSent' => [
+    'Bytes sent.', ['sent_bytes'], ['byte_size'], ['protocol']
+  ],
```
Comment (bruceg) [inline scripts/check-events:22]: I'm surprised this works. These items have different numbers of elements in the value list.

File: scripts/check-events [hunk 4]
```diff
@@ -38,48 +58,72 @@ class Event
...
+        tags.each do |tag, value|
+          if tag == 'stage'
+            if !value.start_with? 'error_stage::'
+              @reports.append("Counter \"#{name}\" tag \"#{tag}\" value must be an \"error_stage\" constant.")
+            end
+          elsif tag == 'error_type'
+            if !value.start_with? 'error_type::'
+              @reports.append("Counter \"#{name}\" tag \"#{tag}\" value must be an \"error_type\" constant.")
+            end
+          end
```
Comment (bruceg) [inline scripts/check-events:125]: This appears to have dropped the check on `error_code` being constant. Is that not a useful check?

**Review by bruceg — COMMENTED** [review]
The changes to the individual events are good, but I have some questions about `check-events` itself.

File: scripts/check-events (hunk 3 above)
Comment (jdrouet) [reply scripts/check-events:22]: ruby is... surprising...

File: scripts/check-events (hunk 4 above)
Comment (jdrouet) [reply scripts/check-events:125]: As we concluded when discussing with @jszwedko, `error_code` might not always be constants. For example, when being build out of a http status code, we end up not having a constant but something like `format!("http_status_{}", status_code)`. That's why strictly forcing it to be a constant doesn't work.

File: scripts/check-events (hunk 1 above)
Comment (jdrouet) [reply scripts/check-events:9]: May I ask you why we'd need to structs to have those fields?
IMO, we need them just to use them as tags in the logs and metrics, and therefore we don't really require them in the structs as long as we have them in the tags.
But if I'm wrong and having those fields in the structs bring a real value, feel free to tell me.

File: scripts/check-events (hunk 4 above)
Comment (bruceg) [reply scripts/check-events:125]: Ah, right, got it.

File: scripts/check-events (hunk 1 above)
Comment (bruceg) [reply scripts/check-events:9]: I guess you're right, the important detail is that the output logs and counters have the right fields. They can be "virtual" in the struct if they are implied by the context.

File: scripts/check-events (hunk 3 above)
Comment (bruceg) [reply scripts/check-events:22]: This will need to be corrected still.

File: src/internal_events/http.rs [hunk 5]
```diff
@@ -94,7 +99,7 @@ impl<'a> InternalEvent for HttpBadRequest<'a> {
     fn emit_metrics(&self) {
         counter!(
             "component_errors_total", 1,
-            "error_code" => self.error_code(),
+            "error_code" => self.error_code.clone(),
```
Comment (bruceg) [inline src/internal_events/http.rs:102]: FTR this kind of thing could be avoided with #9593

File: scripts/check-events (hunk 3 above)
Comment (jdrouet) [reply scripts/check-events]: done!

Comment (jdrouet) [reply scripts/check-events]: my bad

**Review by bruceg — CHANGES_REQUESTED** [review]

File: scripts/check-events [hunk 6]
```diff
@@ -6,12 +6,32 @@ require 'find'
...
+    'Bytes received.', ['received_bytes'], ['byte_size']
+  ],
+  'EventsReceived' => [
+    'Events received.', ['received_events', 'received_event_bytes'], ['count', 'byte_size']
+  ],
+  'EventsSent' => [
+    'Events sent.', ['sent_events', 'sent_event_bytes'], ['count', 'byte_size']
+  ],
+  'BytesSent' => [
+    'Bytes sent.', ['sent_bytes'], ['byte_size']
```
[the remark this answers is not in the document]
Comment (jdrouet) [reply scripts/check-events]: you're right, I don't know why I dropped it 🤔

**Review by bruceg — APPROVED** [review]

File: src/sources/aws_sqs/source.rs (hunk 2 above)
Comment (jszwedko) [reply src/sources/aws_sqs/source.rs:86]: Should this also be an internal event?

**Pull request merged**